MyCryptoHQ/MyCrypto
# Revamp Offline Status Checker

Closes #1915

### Changes

* Change offline detection method to be event emitter based, so it instantly reacts to offline states 

### Steps to Test

1. Load app while online
2. Online indicator should be green
3. Go to dev tools
4. Check `offline` box
5. App should go offline, but it'll do a check to see if it can reach any nodes it's connected to
6. If it can reach a node, itll go back online
7. Uncheck `offline` box, app should still remain online if steps 6 completed successfully
8. Completely disable your network connection 
9. App should go offline, node checks should fail, so app should stay offline while network connection is disabled
10. Re-enable your network connection
11. After a short period, if a node is online, app should regain connection to network and display being connected back to network
12. Login with wallet
13. Notice that nonce field is highlighted with a red border and is visible immediately.
